pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -0,0 +1,455 @@
+	if len(lo.Sort.PrimaryField) > 0 {
+		columnName := toColumnName(lo.Sort.PrimaryField)
+		direction := "ASC"
+		if lo.Sort.PrimaryOrder == DESC {
+			direction = "DESC"
+		}
+		orderByClauses = append(orderByClauses, fmt.Sprintf(`f."%s" %s`, columnName, direction))
+	}
+	if len(lo.Sort.SecondaryField) > 0 {
+		columnName := toColumnName(lo.Sort.SecondaryField)
+		direction := "ASC"
+		if lo.Sort.SecondaryOrder == DESC {
+			direction = "DESC"
+		}
+		orderByClauses = append(orderByClauses, fmt.Sprintf(`f."%s" %s`, columnName, direction))
+	}
+
+	if len(orderByClauses) > 0 {
+		stmt += "\n  ORDER BY "
+		stmt += strings.Join(orderByClauses, ", ")
+	} else {
+		// make sure one default order is always picked
+		if l.namespaced {
+			stmt += "\n  ORDER BY f.\"metadata.namespace\" ASC, f.\"metadata.name\" ASC "
+		} else {
+			stmt += "\n  ORDER BY f.\"metadata.name\" ASC "
+		}
+	}
+
+	// 3- Pagination: LIMIT clause (from lo.Pagination and/or lo.ChunkSize/lo.Resume)
+	limitClause := ""
+	offsetClause := ""
+	// take the smallest limit between lo.Pagination and lo.ChunkSize
+	limit := lo.Pagination.PageSize
+	if limit == 0 || (lo.ChunkSize > 0 && lo.ChunkSize < limit) {
+		limit = lo.ChunkSize
+	}
+	if limit > 0 {
+		limitClause = "\n  LIMIT ?"
+		// note: retrieve one extra row. If it comes back, then there are more pages and a continueToken should be created
+		params = append(params, limit+1)
+	}
+
+	// OFFSET clause (from lo.Pagination and/or lo.Resume)
+	offset := 0
+	if lo.Resume != "" {
+		offsetInt, err := strconv.Atoi(lo.Resume)
+		if err != nil {
+			return nil, "", err
+		}
+		offset = offsetInt
+	}
+	if lo.Pagination.Page >= 1 {
+		offset += lo.Pagination.PageSize * (lo.Pagination.Page - 1)
+	}
+
+	if offset > 0 {
+		offsetClause = "\n  OFFSET ?"
+		params = append(params, offset)
+	}
+
+	stmt += limitClause
+	stmt += offsetClause
+
+	// log the final query
+	logrus.Debugf("ListOptionIndexer prepared statement: %v", stmt)
+	logrus.Debugf("Params: %v", params...)
+
+	// execute
+	prepStmt := l.Prepare(stmt)
+	defer l.CloseStmt(prepStmt)
+	rows, err := l.QueryForRows(ctx, prepStmt, params...)
+	if err != nil {
+		return nil, "", err
+	}
+	items, err := l.ReadObjects(rows, l.GetType(), l.GetShouldEncrypt())
+	if err != nil {
+		return nil, "", err
+	}
+
+	continueToken := ""
+	if limit > 0 && len(items) == limit+1 {
+		// remove extra row
+		items = items[:limit]
+		continueToken = fmt.Sprintf("%d", offset+limit)
+	}
+
+	return toUnstructuredList(items), continueToken, nil
+}
+
+/* Utilities */
+
+// buildORClause creates an SQLite compatible query that ORs conditions built from passed filters
+func buildORClause(orFilters OrFilter) (string, []any) {
+	var orWhereClause string
+	var params []any
+
+	for index, filter := range orFilters.Filters {
+		opString := "LIKE"
+		if filter.Op == NotEq {
+			opString = "NOT LIKE"
+		}
+		columnName := toColumnName(filter.Field)
+		orWhereClause += fmt.Sprintf(`f."%s" %s ?`, columnName, opString)
```

Comment:
The column name seems to be end-user provided input. This seems like a case for injection to me. Can you give this a look and let me know what you think?

Reply:
Same as https://github.com/rancher/lasso/pull/65#discussion_r1612435288.